overlord/managers_test.go

```diff
@@ -5849,3 +5862,164 @@ type: kernel`
 		"snap_try_kernel": "",
 	})
 }
+
+type gadgetUpdatesSuite struct {
+	baseMgrsSuite
+}
+
+var _ = Suite(&gadgetUpdatesSuite{})
+
+func (ms *gadgetUpdatesSuite) SetUpTest(c *C) {
+	ms.baseMgrsSuite.SetUpTest(c)
+
+	bloader := boottest.MockUC16Bootenv(bootloadertest.Mock("mock", c.MkDir()))
+	bootloader.Force(bloader)
+	ms.AddCleanup(func() { bootloader.Force(nil) })
+
+	restore := release.MockOnClassic(false)
+	ms.AddCleanup(restore)
+
+	mockServer := ms.mockStore(c)
+	ms.AddCleanup(mockServer.Close)
+
+	st := ms.o.State()
+	st.Lock()
+	defer st.Unlock()
+
+	// setup model assertion
+	model := ms.brands.Model("can0nical", "my-model", modelDefaults, map[string]interface{}{
+		"gadget": "pi",
+	})
+	devicestatetest.SetDevice(st, &auth.DeviceState{
+		Brand:  "can0nical",
+		Model:  "my-model",
+		Serial: "serialserial",
+	})
+	err := assertstate.Add(st, model)
+	c.Assert(err, IsNil)
+}
+
+func (ms *gadgetUpdatesSuite) makeMockedDev(c *C, structureName string) {
+	// mock /dev/disk/by-label/{structureName}
+	byLabelDir := filepath.Join(dirs.GlobalRootDir, "/dev/disk/by-label/")
+	err := os.MkdirAll(byLabelDir, 0755)
+	c.Assert(err, IsNil)
+	// create fakedevice node
+	err = ioutil.WriteFile(filepath.Join(dirs.GlobalRootDir, "/dev/fakedevice0p1"), nil, 0644)
+	c.Assert(err, IsNil)
+	// and point the mocked by-label entry to the fakedevice node
+	err = os.Symlink(filepath.Join(dirs.GlobalRootDir, "/dev/fakedevice0p1"), filepath.Join(byLabelDir, structureName))
+	c.Assert(err, IsNil)
+
+	// mock /proc/self/mountinfo with the above generated paths
+	ms.AddCleanup(osutil.MockMountInfo(fmt.Sprintf("26 27 8:3 / %[1]s/%s rw,relatime shared:7 - vfat %[1]s/dev/fakedevice0p1 rw", dirs.GlobalRootDir, structureName)))
```

Comment:
Happy to update the mocked mount info to take the `%[2]s` - as for the mount point: The code that searches for the this findMountPointForStructure() and it loads the mountinfo and tries to find the mount point. However you are correct that it does not have to be under "/", it can really be anything. I can make it /run/mnt/{structure-name} in the mock? Or /boot/{structure-name} ? And/or add a comment? But it's important that there is a mount point here or the gadget update code will not be happy.

Reply:
ah right we still need the structures to be mounted somewhere for the code in updaterForStructureImpl where the structure has a filesystem. in this case maybe something specific like `/run/mn/t{structutre-name}` would be more clear to me

Reply:
Thanks! I updated the code now to take these suggestions into account.